csharp/LogicalReadConverterFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+using System;
+
+namespace ParquetSharp
+{
+    /// <summary>
+    /// Extendable class that handles the mapping between a Parquet physical type and the C# logical type when reading values.
+    /// </summary>
+    public class LogicalReadConverterFactory
+    {
+        /// <summary>
+        /// Return a reader delegate if a TPhysical column reader can directly write into a TLogical span (e.g. float to float, int to uint, etc).
+        /// Otherwise return null. This is an optimisation to avoid needless memory copies between buffers (i.e. otherwise we have to use the
+        /// identity converter).
+        /// </summary>
+        public virtual Delegate? GetDirectReader<TLogical, TPhysical>() 
```

Comment:
Why use an untyped Delegate as the return type here and elsewhere in the public API? It's not very obvious for users inheriting from this class what exactly they should be returning without digging into the code.

Reply:
Surprisingly makes the code a lot clearer elsewhere. See the diffs between `LogicalRead.cs` (same for write):

`return (DirectReader) (Delegate) (LogicalRead<TPhysical, TPhysical>.DirectReader) Read;` 
vs
`return LogicalRead.GetDirectReader<TPhysical, TPhysical>();`

The problem is due to how C# generics are handled, you always need to cast to `Delegate` before you can recast up. The new code now only does the re-cast to the type-safe delegate in one place, and the factories are okay to return `Delegate` for code simplicity. Old code had to do that cast everywhere otherwise.

Reply:
OK that makes sense, it would be good to have the expected types documented in the XML docs for LogicalReadConverterFactory and LogicalWriteConverterFactory though to make this a bit clearer for users (eg. maybe just a reference to the typed delegates).

Reply:
Done